src/realm/table.cpp

```diff
@@ -966,6 +966,9 @@ void Table::add_search_index(ColKey col_key, IndexType type)
     auto spec_ndx = leaf_ndx2spec_ndx(col_key.get_index());
     auto attr = m_spec.get_column_attr(spec_ndx);
 
+    if (col_key == m_primary_key_col && type == IndexType::Fulltext)
```

Comment:
by the same logic, should we prevent users from removing an index on a primary key column?

Reply:
also what should set_primary_key_column also check for this combination?

Reply:
@kiburtse yes you are right, if there is a full text search index we can't set the column as pk. 
@ironage yes, we should not allow the user to remove an index for the pk.

But we need also some testing.